pkg/apb/executor.go

```diff
@@ -80,27 +81,28 @@ func ExecuteApb(
 	if err != nil {
 		return executionContext, err
 	}
-	if len(secrets) > 0 {
-		executionContext.Namespace = clusterConfig.Namespace
-		executionContext.Targets = append(executionContext.Targets, context.Namespace)
-	} else {
-		// Using a new UUID is sane, because it will have some gurantee
-		// of uniquenes and will meet DNS name requirements.
-		executionContext.Namespace = uuid.New()
-		executionContext.Targets = append(executionContext.Targets, context.Namespace)
-		// Create namespace.
-		namespace := v1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Labels: map[string]string{transientNameSpaceKey: spec.FQName},
-				Name:   executionContext.Namespace,
-			},
-		}
-		_, err := k8scli.CoreV1().Namespaces().Create(&namespace)
-		if err != nil {
-			return executionContext, err
-		}
+
+	// Using a new UUID is sane, because it will have some gurantee
+	// of uniquenes and will meet DNS name requirements.
+	executionContext.Namespace = uuid.New()
+	executionContext.Targets = append(executionContext.Targets, context.Namespace)
+	// Create namespace.
+	namespace := v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{transientNameSpaceKey: spec.FQName},
+			Name:   executionContext.Namespace,
```

Comment:
You could use `GenerateName`

https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#idempotency, but then you would have to get the name after you have created the namespace.

Reply:
I think then we could give the namespace a name that is more obvious like apb-namespace-(unique suffix).